src/common/src/util/resource_util.rs

```diff
@@ -0,0 +1,322 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Container aware resource utility functions.
+pub mod resource_util {
+    use std::path::Path;
+    use std::fs;
+    use std::env;
+
+    pub enum CgroupVersion {
+        V1,
+        V2,
+    }
+
+    // Current controllers available in immplementation.
+    pub enum Controller {
+        Cpu,
+        Memory,
+    }
+
+    // Default constant Cgroup paths and hierarchy.
+    const DEFAULT_CGROUP_ROOT_HIERARCYHY:  &str = "/sys/fs/cgroup";
+    const DEFAULT_CGROUP_V2_CONTROLLER_LIST_PATH: &str = "/sys/fs/cgroup/cgroup.controllers";
+    const DEFAULT_DOCKER_ENV_PATH: &str = "/.dockerenv";
+    const DEFAULT_KUBERNETES_SECRETS_PATH: &str = "/var/run/secrets/kubernetes.io";
+    const DEFAULT_LINUX_IDENTIFIER: &str = "linux";
+    const DEFAULT_IN_CONTAINER_ENV_VARIABLE: &str = "IN_CONTAINER";
+
+    pub fn get_default_root_hierarchy() -> String {
+        String::from(DEFAULT_CGROUP_ROOT_HIERARCYHY)
+    }
+
+    fn is_linux_machine() -> bool {
+        if env::consts::OS.eq(DEFAULT_LINUX_IDENTIFIER){
+            return true;
+        }
+        false
+    }
+
+    // returns a cgroup version if it exists, else returns None.
+    // Checks for the existence of the root hierarchy directory.
+    fn get_cgroup_version() -> Option<CgroupVersion> {
+        // check if cgroup exists.
+        if !Path::new(DEFAULT_CGROUP_ROOT_HIERARCYHY).is_dir(){
+            return None
+        }
+        // if cgroup.controllers exist, v2 is used.
+        if Path::new(DEFAULT_CGROUP_V2_CONTROLLER_LIST_PATH).exists(){
+            return Some(CgroupVersion::V2)
+        }else{
+            return Some(CgroupVersion::V1)
+        }
+    }
+
+    // checks if is running in a docker container by checking for docker env file, or if it is running in a kubernetes pod.
+    pub fn is_running_in_container() -> bool {
```

Comment:
nit: not necessary to be public if we want to put the function in `resource_util`. If it is intended to provide a function to let the process know if it's running inside the container, we should separate this into another module, such as `runtime`.

```suggestion
    fn is_running_in_container() -> bool {
```